detection_rules/rule.py

```diff
@@ -909,6 +910,15 @@ def validates_esql_data(self, data, **kwargs):
         if data.get('index'):
             raise ValidationError("Index is not a valid field for ES|QL rule type.")
 
+        # check non-aggregate queries have metadata
+        metadata_pattern = r'from\s+\S+\s+metadata\s+_id,\s*_version,\s*_index'
+        stats_pattern = r'\bstats\b.*\bby\b'
+        query = data['query'].lower()
+        if not re.search(stats_pattern, query) and not re.search(metadata_pattern, query):
```

Comment:
nit:

We could `re.compile` these two regex into a single pattern. This would allow use to use one `re.search` (which in the long run may help with performance).